src/demo/text-demo.ts

```diff
@@ -78,108 +101,75 @@ function main() {
   // Configuration option for dat.GUI settings.
   const settings = {
     transitionTimeMs: 250,
+    corpusOffset: 0,
+    corpusPercentage: 10,
     borderPlacement: 1,
     borderRadiusRelative: .1,
     maxBatchTimeMs: 20,
     align: 'left' as AlignmentOption,
     verticalAlign: 'middle' as VerticalAlignmentOption,
-    sort: 'ascending',
-    facet: 'type',
     borderColor: '#000000',
     fillColor: '#ffffff',
     clearBeforeUpdate: false,
   };
 
-  // As a synthetic data set, use the properties on the Window object.
-  const win = window as unknown as {[key: string]: unknown};
-
-  interface Property {
-    name: string;
-    type: string;
+  interface Word {
+    row: number;
+    col: number;
+    start: number;
+    end: number;
   }
 
-  const properties: Property[] = Object.keys(win).map((key) => {
-    return {
-      name: key,
-      type: typeof win[key],
-    };
-  });
+  let col = 0;
+  let row = 0;
+  let word: Word|undefined = undefined;
+  const corpus: Word[] = [];
+
+  for (let pos = 0; pos < TEXT.length; pos++) {
+    const ch = TEXT.charAt(pos);
+
+    if (/\n/.test(ch)) {
+      row++;
+      col = 0;
+    }
+
+    if (!word) {
+      if (/\S/.test(ch)) {
+        word = {col, row, start: pos, end: NaN};
+        corpus.push(word);
+      }
+    } else if (/\s/.test(ch)) {
+      word.end = pos;
+      word = undefined;
+    }
 
-  interface Label {
-    x: number;
-    y: number;
-    text: string;
-    property: Property;
+    col++;
   }
 
-  const labels = properties.map((property) => {
-    return {
-      x: 0,
-      y: 0,
-      text: `${property.name}: ${property.type}`,
-      property,
-    };
-  });
+  if (word) {
+    word.end = TEXT.length;
+  }
 
-  const textSelection = scene.createTextSelection<Label>();
+  const textSelection = scene.createTextSelection<Word>();
```

Comment:
Worth noting that a difference between this demo and the behavior I'm seeing is that I have initialize a `TextSelection<string>()`. Not sure if/how that might affect the internals.

Reply:
Yeah, I noticed that. I think a lot of the issues have to do with misunderstanding of the timing of the Selection callbacks. Hoping to remediate that root cause shortly, per offline conversation.